### PR TITLE
KZOO-120: Add carrier ACL when a local carrier is added.

### PR DESCRIPTION
**Pull request by holy-batman**

### CODE REVIEW

File: src/modules/cb_resources.erl [hunk 1]
```diff
@@ -300,15 +294,14 @@ validate_jobs(Context, ?HTTP_PUT) ->
 
 -spec post(cb_context:context(), path_token()) -> cb_context:context().
 post(Context, ?COLLECTION) ->
-    maybe_reload_acls(cb_context:db_name(Context)),
+    _ = reload_trusted(),
```
Comment (lazedo) [inline src/modules/cb_resources.erl]: reload should be done on success not before

File: src/modules/cb_resources.erl [hunk 2]
```diff
@@ -352,26 +342,24 @@ delete(Context, ResourceId) ->
 %% @end
 %%------------------------------------------------------------------------------
 -spec maybe_aggregate_resource(cb_context:context(), kz_term:ne_binary(), crossbar_status()) -> cb_context:context().
-maybe_aggregate_resource(Context, ?KZ_OFFNET_DB, _) -> Context;
+maybe_aggregate_resource(Context, ?KZ_OFFNET_DB, _) ->
+    _ = reload_trusted(),
```
Comment (lazedo) [inline src/modules/cb_resources.erl:345]: reload should be done `once` on successful completion

File: src/modules/cb_resources.erl [hunk 3]
```diff
@@ -318,13 +311,11 @@ patch(Context, Id) -> do_post(Context, Id).
 -spec put(cb_context:context()) -> cb_context:context().
 put(Context) ->
     Db = cb_context:db_name(Context),
-    maybe_reload_acls(Db),
     Context1 = crossbar_doc:save(Context),
     maybe_aggregate_resource(Context1, Db, cb_context:resp_status(Context1)).
 
 -spec put(cb_context:context(), path_token()) -> cb_context:context().
 put(Context, ?COLLECTION) ->
-    maybe_reload_acls(cb_context:db_name(Context)),
```
Comment (lazedo) [inline src/modules/cb_resources.erl:327]: processing a collection the way it is now would `reload` the number of the docs

**Review by lazedo — CHANGES_REQUESTED** [review]
reload should be done on successful completion, not on each doc (in case of collection)

File: src/modules/cb_resources.erl (hunk 2 above)
Comment (holy-batman) [reply src/modules/cb_resources.erl:345]: changed to only refresh if success.

File: src/modules/cb_resources.erl (hunk 3 above)
Comment (holy-batman) [reply src/modules/cb_resources.erl:327]: changed collection refresh to be done after a success is received from collection.

**Review by lazedo — APPROVED** [review]

File: src/modules/cb_resources.erl [hunk 4]
```diff
@@ -652,119 +516,21 @@ aggregate_resource(ResourceJObj) ->
...
-%%------------------------------------------------------------------------------
-%% @doc
-%% @end
-%%------------------------------------------------------------------------------
--spec maybe_remove_aggregates(kzd_resources:docs()) -> 'ok'.
-maybe_remove_aggregates([]) -> 'ok';
-maybe_remove_aggregates([ResourceJObj|ResourceJObjs]) ->
-    _ = remove_aggregate(kz_doc:id(ResourceJObj)),
-    maybe_remove_aggregates(ResourceJObjs).
+    _ = aggregate_resource([ResourceJObj]),
```
Comment (jamesaimonetti) [inline src/modules/cb_resources.erl]: Why wrap this in a list? Shouldn't `aggregate_resource/1` take a JSON object only, not a list of JSON objects?

And I'd rather just call `lists:foreach/2` here:

```erlang
maybe_aggregate_resources(ResourceJObjs) ->
    lists:foreach(fun aggregate_resource/1, ResourceJObjs).
```

**Review by jamesaimonetti — CHANGES_REQUESTED** [review]

File: src/modules/cb_resources.erl (hunk 4 above)
Comment (holy-batman) [reply src/modules/cb_resources.erl]: @jamesaimonetti fixed :)

**Pull request merged**